stdlib/public/stubs/UnicodeNormalization.cpp

```diff
@@ -36,6 +36,11 @@
 
 #pragma clang diagnostic pop
 
+// clang 4.0 under a recent linux has unicode with built-in char16_t support
+#if (defined(__CYGWIN__) || defined( _MSC_VER)) || defined(__linux__)
+#define __USE_UCHAR 1
```

Comment:
I've update the pull request here: https://github.com/apple/swift/pull/10361 

Had to delete the original branch, since I managed to mess it up - I didn't keep an upstream-master branch without any changes syncd.